src/screens/hardware-wallets/PairHardwareWalletAgainSheet.tsx

```diff
@@ -23,12 +23,35 @@ import { useTheme } from '@/theme';
 import { IS_IOS } from '@/env';
 import { Layout } from '@/screens/hardware-wallets/components/Layout';
 import { TRANSLATIONS } from '@/screens/hardware-wallets/constants';
+import { useRecoilState } from 'recoil';
+import {
+  HARDWARE_TX_ERROR_KEY,
+  LedgerIsReadyAtom,
+  ledgerStorage,
+  readyForPollingAtom,
+} from '@/navigation/HardwareWalletTxNavigator';
+import { TryAgainButton } from './components/TryAgainButton';
+import { useMMKVBoolean } from 'react-native-mmkv';
 
 const INDICATOR_SIZE = 7;
 
 export const PairHardwareWalletAgainSheet = () => {
   const { isDarkMode } = useTheme();
-  const connected = false;
+
+  const [isReady, setIsReady] = useRecoilState(LedgerIsReadyAtom);
+  const [readyForPolling, setReadyForPolling] = useRecoilState(
+    readyForPollingAtom
+  );
+  const [hardwareTXError, setHardwareTXError] = useMMKVBoolean(
+    HARDWARE_TX_ERROR_KEY,
+    ledgerStorage
+  );
```

Comment:
the `useMMKVAnything` family of hooks is not great. The implementation uses some mmkv event listeners and when you set it the component will re-render in the next cycle and it probably won't get batched with the rest of normal React state updates. We can discuss the use case and see if there's a better option.

Reply:
It was causing problems in the `usePersistentDominantColor` hook for sure

Reply:
Whaaaaat I remember these now. Yeah that'd be a nice thing to refactor one day.